src/main/java/com/sourcegraph/vcs/RepoUtil.java

```diff
@@ -47,7 +49,7 @@ public static RepoInfo getRepoInfo(@NotNull Project project, @NotNull VirtualFil
         relativePath = relativePath.substring(relativePath.indexOf("/") + 1);
       }
 
-      remoteUrl = getRemoteRepoUrl(project, file);
+      remoteUrl = getRemoteRepoUrl(project, file).join(); // todo
```

Comment:
I don't have practical experience with Java CompletableFutures, but it seems to me that this code will be blocking no matter how long we postpone using `join` in this case.

IMO that's totally fine, but what would be the difference if we used join immediately after sending a request to the server (for example: `server.convertGitCloneURLToCodebaseName(cloneURL).join()`)? It looks to me that the effect would be the same but code would be simpler. I'm not sure if there's room for writing asynchronous code here since this logic heavily rely on the agent's response.

Reply:
Thanks! Very good point. It looks like this code is run on a pooled thread anyway and it is safe to `.join()` immediately. Everything is simple now 😅